activerecord/lib/active_record/relation/predicate_builder.rb

```diff
@@ -87,14 +87,14 @@ def create_binds_for_hash(attributes)
         binds = []
 
         attributes.each do |column_name, value|
-          case value
-          when Hash
+          case
+          when value.is_a?(Hash)
```

Comment:
@esparta Thank you for your suggestion. I changed to keep the case statement. Thanks!